src/sources/datadog_agent/logs.rs

```diff
@@ -192,3 +201,89 @@ pub(crate) fn decode_log_body(
 
     Ok(decoded)
 }
+
+// ddtags input is a string containing a list of key-value pairs.
+// the tag-value list members are separated by `,` and the
+// tag-value pairs are separated by `:`.
+// The output is an Object constructed with the tag-value pairs.
+//
+// Technically, the Datadog logs intake accepts input where this
+// field is a bare tag (no `:`), so we're supporting that here,
+// in which case the raw input is not expanded into an object.
+fn parse_ddtags(ddtags_raw: &Bytes) -> Value {
+    let ddtags_str = String::from_utf8_lossy(ddtags_raw);
+
+    // The value is either an empty string or a bare tag
+    if !ddtags_str.contains(',') && !ddtags_str.contains(':') {
+        return Value::Bytes(ddtags_raw.clone());
+    }
```

Comment:
Should this result in just the string (`Bytes`) or something like `{"TAG":null}`?

Reply:
Originally I had the later. But then I switched to the former because It seemed odd to have an object with a single entry that contains a key-value pair where the value is null 🤷  Essentially the way I have it means the input 

`"ddtags" : "foo"`

, isn't translated. 

This definitely seems to be an edge case but I'll try to see if there is behavior I can discern from the DD logs intake to mimic.

Reply:
Here's my thinking: Would all possible combinations of structured tags be able to be converted into a ddtags string and then parsed back into the same structure? OTOH it's worth asking how the Datadog logs intake handles them and act according to those expectations.

Reply:
Running some experiments locally, I saw that the logs backend accepts a string that contains either:

1. a single bare tag
2. an empty string (no ddtags are parsed)
3. a single or multi key-values
4. a combination of bare tag and key-values

After thinking about it some more, here is my proposal on how to handle each case, which is a blend of the DD logs backend behavior, and considering what this option in Vector is doing, and the existing log namespacing for this component:

In each case, we always have a `ddtags` JSON object. I think this is just good practice to have predictable consistent behavior. We can clearly state in the docs/guides that if the `ddtags` log field is present and is a string, when this option is set, that string value is now an Object instead.

1. The object contains a single entry `{"baretag": NULL}`
2. The object is added but is empty. `{}`
3. (self explanatory, k-v pair(s))
4. (same as 3 except that bare tags are represented as in # 1)

WDYT?

Reply:
I agree with that approach, yes.

Reply:
Addressed in the latest commit and updated the code comments accordingly